Mage/src/main/java/mage/abilities/keyword/AnnihilatorAbility.java

```diff
@@ -123,9 +124,9 @@ public boolean apply(Game game, Ability source) {
             player = game.getPlayer(defendingPlayerId);
         }
         if (player != null) {
-            int amount = Math.min(count, game.getBattlefield().countAll(StaticFilters.FILTER_CONTROLLED_PERMANENT, player.getId(), game));
+            int amount = Math.min(count, game.getBattlefield().countAll(new FilterControlledPermanent(), player.getId(), game));
             if (amount > 0) {
-                Target target = new TargetControlledPermanent(amount, amount, StaticFilters.FILTER_CONTROLLED_PERMANENT, true);
+                Target target = new TargetControlledPermanent(amount, amount, new FilterControlledPermanent(), true);
```

Comment:
Why that's broke all other tests? Static filter value disposing somehow in this class?

Reply:
That was really weird. StaticFilters.FILTER_CONTROLLED_PERMANENT was changed somehow during tests (added weird predicates which leaded to ability issue). Generally I think that idea of StaticFilters class fields is just wrong... Even if we set fields as final it doesn't protected referenced objects from being modified. That's pretty bad.

Reply:
Yep, I've found that. Look at my post here:  #4402